### DISCUSSION TITLE
Bond cleavage (ReactionFromSmarts) without adding hydrogen atoms

### DISCUSSION THREAD

**Discussion by JunYang2021** [rdkit/rdkit#5906] [Q&A] [answered]
Hello, I want to use `ReactionFromSmarts` to represent a C-C bond cleavage, which is like the reaction in this picture. 
<img width="676" alt="2" src="https://user-images.githubusercontent.com/88775486/209814626-addffd61-e5d6-45b1-b7bf-275cb90be302.png">
I write the reaction: 
```
cc_single_bond_break = AllChem.ReactionFromSmarts('[C:1][C:2]>>[C:1].[C:2]')
products = cc_single_bond_break.RunReactants((c_chain,))
```
However, then hydrogens will automatically be added on C, rather than generating radicals. What should I do to react without adding hydrogens?

**Comment by greglandrum** [comment, marked as answer]
@JunYang2021 There's no easy way to express that using Reaction SMARTS (you can do it with RXN if you want), but enough information is available for you to do with you want with a bit of post-processing code.

The products that you get from `RunReactants` have some properties set which provide you all the information you need. The two important ones here are `_ReactionDegreeChanged`, which is set on atoms where the degree (number of bonds) changed in the reaction, and `react_atom_idx`, which gives you the index of the atom in the reacting molecule.

Those are used in this function:
```
def add_radicals_for_broken_bonds(product,reactant):
    for atom in product.GetAtoms():
        if atom.HasProp('_ReactionDegreeChanged') and atom.GetIntProp('_ReactionDegreeChanged'):
            newVal = 0
            for bond in atom.GetBonds():
                newVal += bond.GetValenceContrib(atom)
            oldVal = reactant.GetAtomWithIdx(atom.GetIntProp('react_atom_idx')).GetExplicitValence()
            atom.SetNumRadicalElectrons(int(oldVal-newVal))
```

Which can be used together with your reaction:
```
c_chain = Chem.MolFromSmiles('CCCC')
cc_single_bond_break = AllChem.ReactionFromSmarts('[C:1][C:2]>>[C:1].[C:2]')
products = cc_single_bond_break.RunReactants((c_chain,))
prod_copy = Chem.Mol(products[1][0])
add_radicals_for_broken_bonds(prod_copy,c_chain)
Chem.SanitizeMol(prod_copy)
prod_copy.Debug()
```
The output of that is:
```
Atoms:
	0 6 C chg: 0  deg: 1 exp: 1 imp: 2 hyb: 4 arom?: 0 chi: 0 rad: 1
	1 6 C chg: 0  deg: 2 exp: 2 imp: 2 hyb: 4 arom?: 0 chi: 0
	2 6 C chg: 0  deg: 1 exp: 1 imp: 3 hyb: 4 arom?: 0 chi: 0
Bonds:
	0 0->1 order: 1 conj?: 0 aromatic?: 0
	1 1->2 order: 1 conj?: 0 aromatic?: 0
```

**Reply by JunYang2021** [reply]
Thank you for your smart method! This method can exactly solve my problem. 
Before getting your answer, I have solved this by listing all the circumstances of hydrogen numbers, which is a little cumbersome.
